pilot/pkg/networking/util/util.go

```diff
@@ -256,22 +256,43 @@ func ConvertLocality(locality string) *core.Locality {
 		return nil
 	}
 
-	items := strings.Split(locality, "/")
+	region, zone, subzone := splitLocality(locality)
+	return &core.Locality{
+		Region:  region,
+		Zone:    zone,
+		SubZone: subzone,
+	}
+}
+
+func LocalityToString(locality *core.Locality) string {
+	if locality == nil {
+		return ""
+	}
+
+	return locality.Region + "/" + locality.Zone + "/" + locality.SubZone
+}
+
+func LocalityMatch(proxyLocality, ruleLocality string) bool {
+	region, zone, subzone := splitLocality(proxyLocality)
+	ruleRegion, ruleZone, ruleSubzone := splitLocality(ruleLocality)
+	regionMatch := ruleRegion == "*" || region == ruleRegion
+	zoneMatch := ruleZone == "*" || ruleZone == "" || zone == ruleZone
+	subzoneMatch := ruleSubzone == "*" || ruleSubzone == "" || subzone == ruleSubzone
+
+	if regionMatch && zoneMatch && subzoneMatch {
+		return true
+	}
+	return false
+}
+
+func splitLocality(locality string) (region, zone, subzone string) {
```

Comment:
Do you think we can cutdown a lot of unnecessary string processing if we create a Locality object in the model with region, zone, subzone and make all the locality api calls return this object?

Reply:
The advantage is that we parse the locality only once and then pass that object around.

Reply:
Will consider this